Sources/CacheAdvance/CacheReader.swift

```diff
@@ -52,6 +55,18 @@ final class CacheReader {
 
         switch try nextEncodedMessageSpan() {
         case let .span(messageLength):
+            // Check our assumptions before we try to read the message.
+            let endOfMessage = Bytes(startingOffset) + Bytes(MessageSpan.storageLength) + Bytes(messageLength)
+            let startingOffsetIsBeforeEndOfNewestMessageAndDoesNotExceedEndOfNewestMessage = startingOffset < offsetInFileAtEndOfNewestMessage && endOfMessage <= offsetInFileAtEndOfNewestMessage
+            let startingOffsetIsOnOrAfterAfterEndOfNewestMessageAndDoesNotExceedEndOfFile = startingOffset >= offsetInFileAtEndOfNewestMessage && endOfMessage <= maximumBytes
```

Comment:
We've already guarded that `startingOffset` is not equal to the offset at the end of the newest message. https://github.com/dfed/CacheAdvance/blob/0b63b1e881f7ce12ff4a7dbe9661a696201e0da9/Sources/CacheAdvance/CacheReader.swift#L51

I wonder if that's an opportunity to simplify here since the starting offset should never be on the ned of the newest message.

Reply:
I like that. Would help us get rid of the `=` check